addons/website_sale/models/sale_order.py

```diff
@@ -359,6 +350,41 @@ class SaleOrderLine(models.Model):
     linked_line_id = fields.Many2one('sale.order.line', string='Linked Order Line', domain="[('order_id', '!=', order_id)]", ondelete='cascade')
     option_line_ids = fields.One2many('sale.order.line', 'linked_line_id', string='Options Linked')
 
+    def unlink(self):
+        to_rename = []
+        for record in self:
+            if record.linked_line_id:
+                to_rename.append(record.linked_line_id)
+        res = super(SaleOrderLine, self).unlink()
+        for record in to_rename:
+            product = record._get_product_with_context()
+            record.name = record.get_sale_order_line_multiline_description_sale(product)
+        return res
+
+    def create(self, vals):
+        res = super(SaleOrderLine, self).create(vals)
+        for record in self:
+            product = record._get_product_with_context()
+            record.name = record.get_sale_order_line_multiline_description_sale(product)
```

Comment:
I don't think it is OK to do it this way unfortunately. In the backend, if the sales person is setting custom text on that field, it will always be overridden.

Maybe if `name` is in `vals`, it shouldn't be done (and make sure `name` is never in `vals` in the case you need).